chainlink-modsec/libmodsec/internal/verifier/interfaces.go

```diff
@@ -0,0 +1,51 @@
+package verifier
+
+import (
+	"context"
+)
+
+// Reader is an interface for reading the blockchain for new work.
+type Reader interface {
+	// Next returns a channel that will yield the next piece of work to be processed.
+	Next(ctx context.Context) <-chan Work
+
+	// Watch returns a channel that will yield all work to be processed.
+	Watch(ctx context.Context) <-chan Work
+}
+
+// Transformer is an interface that defines how to transform a Work item into
+// the generic modsec.Message format and encoded into a destination specific
+// payload. This interface encapsulates most of the chain agnostic services
+// required by the 1.6 implementation.
+type Transformer interface {
+	Transform(work Work) HandlerPayload
+}
```

Comment:
Encapsulated in the interface above - could make things more generic if we think its useful though.

Reply:
I think its important to split reading and re-encoding. This layer will have chain family specific re-encoding details. So may as well split it up from the blockchain reading so that each has a single responsibility.